packages/h/hyprland/pspec_x86_64.xml

```diff
@@ -0,0 +1,274 @@
+<PISI>
+    <Source>
+        <Name>hyprland</Name>
+        <Homepage>https://hyprland.org/</Homepage>
+        <Packager>
+            <Name>GENEVEE Nicolas</Name>
+            <Email>gnick@orange.fr</Email>
+        </Packager>
+        <License>BSD-3-Clause</License>
+        <PartOf>desktop.hyprland</PartOf>
+        <Summary xml:lang="en">Hyprland is a 100% independent, dynamic tiling Wayland compositor that doesn&apos;t sacrifice on its looks.</Summary>
+        <Description xml:lang="en">Hyprland is a 100% independent, dynamic tiling Wayland compositor that doesn&apos;t sacrifice on its looks. It provides the latest Wayland features, is highly customizable, has all the eyecandy, the most powerful plugins, easy IPC, much more QoL stuff than other compositors and more...
+</Description>
+        <Archive type="binary" sha1sum="79eb0752a961b8e0d15c77d298c97498fbc89c5a">https://sources.getsol.us/README.Solus</Archive>
+    </Source>
+    <Package>
+        <Name>hyprland</Name>
+        <Summary xml:lang="en">Hyprland is a 100% independent, dynamic tiling Wayland compositor that doesn&apos;t sacrifice on its looks.</Summary>
+        <Description xml:lang="en">Hyprland is a 100% independent, dynamic tiling Wayland compositor that doesn&apos;t sacrifice on its looks. It provides the latest Wayland features, is highly customizable, has all the eyecandy, the most powerful plugins, easy IPC, much more QoL stuff than other compositors and more...
+</Description>
+        <PartOf>desktop.hyprland</PartOf>
+        <Files>
+            <Path fileType="executable">/usr/bin/Hyprland</Path>
+            <Path fileType="executable">/usr/bin/hyprctl</Path>
+            <Path fileType="executable">/usr/bin/hyprland</Path>
+            <Path fileType="executable">/usr/bin/hyprpm</Path>
+            <Path fileType="data">/usr/share/bash-completion/completions/hyprctl</Path>
+            <Path fileType="data">/usr/share/bash-completion/completions/hyprpm</Path>
+            <Path fileType="data">/usr/share/fish/vendor_completions.d/hyprctl.fish</Path>
+            <Path fileType="data">/usr/share/fish/vendor_completions.d/hyprpm.fish</Path>
+            <Path fileType="data">/usr/share/hypr/hyprland.conf</Path>
+            <Path fileType="data">/usr/share/hypr/lockdead.png</Path>
+            <Path fileType="data">/usr/share/hypr/lockdead2.png</Path>
+            <Path fileType="data">/usr/share/hypr/wall0.png</Path>
+            <Path fileType="data">/usr/share/hypr/wall1.png</Path>
+            <Path fileType="data">/usr/share/hypr/wall2.png</Path>
+            <Path fileType="man">/usr/share/man/man1/Hyprland.1</Path>
+            <Path fileType="man">/usr/share/man/man1/hyprctl.1</Path>
+            <Path fileType="data">/usr/share/wayland-sessions/hyprland.desktop</Path>
+            <Path fileType="data">/usr/share/xdg-desktop-portal/hyprland-portals.conf</Path>
```

Comment:
this is the content of the file : 
```
[preferred]
default=hyprland;gtk
```
does that mean that I have to add this items to the package file for hyprland or that we have to add hyprland to the gtk package ?

Reply:
So the way that XDG portal implementations work is that at runtime the XDG library looks up the file `/usr/share/xdg-desktop-portal/$i-portals.conf`, where `$i` is the current value of the `XDG_CURRENT_DESKTOP` environmental variable, lower-cased. This variable is set by the compositor once you login. Since there are many different portals not every XDG implementation bothers to implement all of them. XDG will check to see if a given implementation supports the portal it's trying to use, and if not it will try the next one in sequence. 

So in this case XDG will check the hyprland portal for everything, and then fall through to the GTK portal if something isn't implemented (file pickers are a very common one that most implementations just leave to the GTK one). My preference here is that the `hyprland` package has a rundep on `xdg-desktop-portal-hyprland` and `xdg-desktop-portal-gtk`. That will give the correct behavior.